frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/ConditionalAccess.tsx

```diff
@@ -0,0 +1,352 @@
+  // to use useQuery's
+  // "error" state handled by onError callback
+  // success state handled by onSuccess callback
+  useQuery<
+    ConfirmMSConditionalAccessResponse,
+    Error,
+    ConfirmMSConditionalAccessResponse
+  >(["confirmAccess"], conditionalAccessAPI.confirmMicrosoftConditionalAccess, {
+    ...DEFAULT_USE_QUERY_OPTIONS,
+    // only make this call at the appropriate UI phase
+    enabled: phase === Phase.ConfirmingConfigured && isPremiumTier,
+    onSuccess: ({ configuration_completed }) => {
+      if (configuration_completed) {
+        setPhase(Phase.Configured);
+        renderFlash(
+          "success",
+          "Successfully verified conditional access integration"
+        );
+      } else {
+        setPhase(Phase.Form);
+        renderFlash(
+          "error",
+          "Could not verify conditional access integration. Please try connecting again."
+        );
+      }
+    },
+    onError: () => {
+      // distinct from successful confirmation response of `false`, this handles an API error
+      setPhase(Phase.ConfirmationError);
+    },
+  });
+
+  const {
+    microsoft_entra_tenant_id: configMsetId,
+    microsoft_entra_connection_configured: configMseConfigured,
+  } = config?.conditional_access || {};
+
+  // only checks if tenant id already present in config, not if user added it to the form
+  useEffect(() => {
+    if (configMsetId) {
+      if (!configMseConfigured) {
+        setPhase(Phase.ConfirmingConfigured);
+      } else {
+        // tenant id is present and connection is configured
+        setPhase(Phase.Configured);
+      }
+    }
+  }, [configMsetId, configMseConfigured]);
+
+  if (!isPremiumTier) {
+    return <PremiumFeatureMessage />;
+  }
+
+  // HANDLERS
+
+  const toggleDeleteConditionalAccessModal = () => {
+    setShowDeleteConditionalAccessModal(!showDeleteConditionalAccessModal);
+  };
+
+  const onSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+
+    const errs = validate(formData);
+    if (Object.keys(errs).length > 0) {
+      setFormErrors(errs);
+      return;
+    }
+    setIsUpdating(true);
+    try {
+      const {
+        microsoft_authentication_url: msAuthURL,
+      } = await conditionalAccessAPI.triggerMicrosoftConditionalAccess(
+        formData[MSETID]
+      );
+      setIsUpdating(false);
+      setPhase(Phase.FormSubmitted);
+      window.open(msAuthURL);
+    } catch (e) {
+      renderFlash(
+        "error",
+        "Could not update conditional access integration settings."
+      );
+      setIsUpdating(false);
+    }
+  };
+
+  const onDeleteConditionalAccess = async () => {
+    setIsUpdating(true);
+    try {
+      await conditionalAccessAPI.deleteMicrosoftConditionalAccess();
+      renderFlash(
+        "success",
+        "Successfully disconnected from Miscrosoft Entra."
+      );
+      setIsUpdating(false);
+      toggleDeleteConditionalAccessModal();
+      setPhase(Phase.Form);
```

Comment:
Nit, we probably should be re-fetching the config here via the API, rather than trusting that this worked (like we do with GitOps mode).  It's pretty low risk here since we should be able to trust our own backend when it says "this action was successful", but always better to be visualizing real data from the backend when we can.

Reply:
Great call. The omission may have been to allow testing with the fake data before the backend
actually works, since if we refetch the backend when it's not actually doing anything it won't lead
to the right phase. Will include this in the "fake data" commit.

Reply:
The goal here is for the page state to be driven by the state of the backend as much as possible.  I realize that for certain parts of the process (like when the user is doing something on another site) we can't easily reach that goal.  But for this case, I think we can by removing the `setPhase(Phase.Form)` here and moving it to the `onSuccess` of the config fetcher (contingent on the new config having `microsoft_entra_connection_configured` be `false`).  Does that make sense?

Reply:
Maybe move the `isUpdating(false)` to there too, just to avoid an edge case of "config load takes a moment and user can do weird things in the meantime"

Reply:
Yes this makes sense